docs/book/src/clusterctl/configuration.md

```diff
@@ -50,6 +50,31 @@ AWS_B64ENCODED_CREDENTIALS: XXXXXXXX
 
 In case a variable is defined both in the config file and as an OS environment variable, the latter takes precedence.
 
+## Overrides Layer
```

Comment:
Being this a user-facing documentation, should we "soft" introduce overrides? e.g 

clusterctl by default reads provider components files, cluster templates files and metadata files from provider repositories.

If you need to override one of the official files with a local version, e.g. for testing purposes, you can use override feature embedded in clusterctl so it will read injected ....

@vincepri opinions^^

Reply:
I'd almost make overrides their own page and have extended docs there, it could also be helpful for folks to find some use cases listed

Reply:
I'm -1 to make a separate page...
WRT to example I think we can add a link to developer page to see an example of how overrides are used

Reply:
Yeah..I'm leaning towards having the information either in the developers.md or here in configuration.md.
I'm ok with adding an example here in this section since we are anyways linking to the developers section regarding running the local overrides hack.

Reply:
Up to you, the overriding behavior as today requires a lot of familiarity with the tool and doesn't give enough use cases for newcomers to understand what they can do with it

Reply:
I can add a little blurb explaining how it can be used.

Reply:
I added some more documentation. Feedback welcome.